spring-pulsar/src/main/java/org/springframework/pulsar/core/PulsarTemplate.java

```diff
@@ -237,14 +244,14 @@ public SendMessageBuilder<T> withTopic(String topic) {
 		}
 
 		@Override
-		public SendMessageBuilder<T> withMessageCustomizer(TypedMessageBuilderCustomizer<T> messageCustomizer) {
-			this.messageCustomizer = messageCustomizer;
+		public SendMessageBuilder<T> withEncryptionKeys(Collection<String> encryptionKeys) {
+			this.encryptionKeys = encryptionKeys;
 			return this;
 		}
 
 		@Override
-		public SendMessageBuilder<T> withCustomRouter(MessageRouter messageRouter) {
```

Comment:
A custom router seems like a common enough setting to leave it surfaced in the builder as convenience and just turn it into the customizer before sending downstream. I also get the motivation of this change, and if we keep that field who is to say that "interceptors" should not be here, oh and what about my other favorite field "foo". 

If we keep it out we need to be sure we clearly call out in the docs "The complicated things you want to do are probably found on the PBC such as custom router, interceptors, etc.."

Reply:
Yes, the MessageRouter doesn't seem more important to me than the other dozen of properties. If we add it, there would be no reason not to add all the other and duplicating the ProducerBuilder niterface.
I would be in favor of documenting how to use PBC to configure the router or the interceptors.

Reply:
done